docs/rules/require-super-in-init.md

```diff
@@ -34,3 +36,9 @@ export default Component.extend({
   }
 });
 ```
+
+## Configuration
+
+This rule takes an optional object containing:
+
+* `boolean` -- `catchInitOnly` -- whether the rule should only check the `init` lifecycle hook and not other lifecycle hooks (default `true`, TODO: change default to `false` and rename rule to `require-super-in-lifecycle-hooks` in next major release)
```

Comment:
The usage of "catch" feels a bit odd here. Maybe something like:

```suggestion
* `string[]` -- `methods` -- the list of methods that the rule should enforce `super` to be called in (default `['init']`, TODO: change default to `['init', 'willDestroy', 'didReceiveAttrs', ....]` and rename rule to `require-super-in-lifecycle-hooks` in next major release)
```

Reply:
@rwjblue that's an interesting idea but I'm not sure if we want/need to provide users with full configurability of the methods list? Having a configurable list would complicate the implementation (since right now we only look for known hooks in specific Ember classes) and make it harder to opt-in to the new functionality (since the user would have to pass a list instead of a boolean). 

My understanding of the desired end-state of this rule is requiring super in:
* `init` in any Ember class
*  any [component lifecycle hook](https://guides.emberjs.com/v3.4.0/components/the-component-lifecycle/) in an Ember component/mixin
* (let me know if I missed anything)

My goal with this PR is to provide transitionary options to help us get there in a non-breaking way:
* `catchInitOnly` (default true)
* `catchNativeClasses` (default false)

Reply:
Ya, that sounds totally reasonable. I was thinking too generic. 😃

I'd still really like to avoid using the word "catch" here to mean anything other than the JavaScript meaning (I find it very confusing).

Reply:
I could replace `catch` with `check`? Although there are currently 5 other options in this plugin that start with `catch`.

Reply:
Updated to use `check` instead.

Reply:
Thank you, sorry for nit picking